File: app/src/main/java/orz/dev/disk/fkmt/ameraam/AmeraamReceiver.java
package orz.dev.disk.fkmt.ameraam;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.support.v4.content.WakefulBroadcastReceiver;
import android.util.Log;

public final class AmeraamReceiver extends WakefulBroadcastReceiver {

    private static final String TAG = AmeraamReceiver.class.getSimpleName();

    @Override
    public void onReceive(Context context, Intent intent) {
        Log.d(TAG, "onReceive");
        Intent ameraam = new Intent(context, AmeraamService.class);
        startWakefulService(context, ameraam);
    }

}

File: app/src/main/kotlin/orz/dev/disk/fkmt/ameraam/AmeraamReceiver.kt
package orz.dev.disk.fkmt.ameraam

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.support.v4.content.WakefulBroadcastReceiver
import android.util.Log
import kotlin.platform.platformStatic

public class AmeraamReceiver : WakefulBroadcastReceiver() {

    override fun onReceive(context: Context, intent: Intent) {
        Log.d(TAG, "onReceive")
        val ameraam = Intent(context, javaClass<AmeraamService>())
        WakefulBroadcastReceiver.startWakefulService(context, ameraam)
    }

    companion object {
        private val TAG = javaClass<AmeraamReceiver>().getSimpleName()
    }

}
